### ISSUE TITLE
Code block formatting takes a long time to execute?

### ISSUE THREAD

**Issue by jerviscui** [platers/obsidian-linter#550]
I add the following to the file and it takes a long time to execute lint, about half a minute.

~~~
the code block: 

```
     Name     |  Owner   | Encoding | Collate | Ctype |   Access privileges   |  Size   | Tablespace |                Description
--------------+----------+----------+---------+-------+-----------------------+---------+------------+-----------------------------
---------------
 OrderModule  | root     | UTF8     | C       | C     |                       | 247 MB  | pg_default |
 PlaceModule  | root     | UTF8     | C       | C     | =Tc/root             +| 207 MB  | pg_default |
              |          |          |         |       | root=CTc/root        +|         |            |
              |          |          |         |       | parking_ro=c/root     |         |            |
 postgres     | postgres | UTF8     | C       | C     |                       | 8141 kB | pg_default | default administrative connection database
```
~~~

**Comment by pjkaufman** [comment]
I am able to see that it takes about 30 seconds to lint the above. Tables are a pain-point and are extremely slow to parse. I have made some changes that make it faster, but I am not sure if that parsing can be made faster. I will see if this is related to the issue with parsing tables not being fast.

**Comment by pjkaufman** [comment]
Here is the timing info for myself:
```
ConsoleHelper.ts:471 [Obsidian Linter] Running linter
ConsoleHelper.ts:471 [Obsidian Linter] Running Format Tags in YAML
ConsoleHelper.ts:471 [Obsidian Linter] Running Escape YAML Special Characters
ConsoleHelper.ts:471 pre rules: 0.767822265625 ms
ConsoleHelper.ts:471 [Obsidian Linter] Running Format Yaml Array
ConsoleHelper.ts:471 format-yaml-array: 0.423828125 ms
ConsoleHelper.ts:471 insert-yaml-attributes: 0.0693359375 ms
ConsoleHelper.ts:471 move-tags-to-yaml: 0.0791015625 ms
ConsoleHelper.ts:471 remove-yaml-keys: 0.063232421875 ms
ConsoleHelper.ts:471 yaml-title: 0.05712890625 ms
ConsoleHelper.ts:471 yaml-title-alias: 0.055908203125 ms
ConsoleHelper.ts:471 capitalize-headings: 0.056884765625 ms
ConsoleHelper.ts:471 file-name-heading: 0.058837890625 ms
ConsoleHelper.ts:471 [Obsidian Linter] Running Header Increment
ConsoleHelper.ts:471 header-increment: 8.219970703125 ms
ConsoleHelper.ts:471 headings-start-line: 0.087158203125 ms
ConsoleHelper.ts:471 remove-trailing-punctuation-in-heading: 0.06201171875 ms
ConsoleHelper.ts:471 footnote-after-punctuation: 0.06201171875 ms
ConsoleHelper.ts:471 move-footnotes-to-the-bottom: 0.06103515625 ms
ConsoleHelper.ts:471 re-index-footnotes: 0.055908203125 ms
ConsoleHelper.ts:471 [Obsidian Linter] Running Auto-correct Common Misspellings
ConsoleHelper.ts:471 auto-correct-common-misspellings: 1.392822265625 ms
ConsoleHelper.ts:471 [Obsidian Linter] Running Convert Bullet List Markers
ConsoleHelper.ts:471 convert-bullet-list-markers: 0.637939453125 ms
ConsoleHelper.ts:471 [Obsidian Linter] Running Emphasis Style
ConsoleHelper.ts:471 emphasis-style: 0.633056640625 ms
ConsoleHelper.ts:471 [Obsidian Linter] Running No Bare URLs
ConsoleHelper.ts:471 no-bare-urls: 0.561767578125 ms
ConsoleHelper.ts:471 ordered-list-style: 0.063720703125 ms
ConsoleHelper.ts:471 [Obsidian Linter] Running Proper Ellipsis
ConsoleHelper.ts:471 proper-ellipsis: 0.6279296875 ms
ConsoleHelper.ts:471 remove-consecutive-list-markers: 0.055908203125 ms
ConsoleHelper.ts:471 [Obsidian Linter] Running Remove Empty List Markers
ConsoleHelper.ts:471 remove-empty-list-markers: 1.677978515625 ms
ConsoleHelper.ts:471 [Obsidian Linter] Running Remove Hyphenated Line Breaks
ConsoleHelper.ts:471 remove-hyphenated-line-breaks: 0.5009765625 ms
ConsoleHelper.ts:471 [Obsidian Linter] Running Remove Multiple Spaces
ConsoleHelper.ts:471 remove-multiple-spaces: 23617.885009765625 ms
ConsoleHelper.ts:471 [Obsidian Linter] Running Strong Style
ConsoleHelper.ts:471 strong-style: 0.523681640625 ms
ConsoleHelper.ts:471 [Obsidian Linter] Running Two Spaces Between Lines with Content
ConsoleHelper.ts:471 two-spaces-between-lines-with-content: 6100.97216796875 ms
ConsoleHelper.ts:471 [Obsidian Linter] Running Unordered List Style
ConsoleHelper.ts:471 unordered-list-style: 0.47216796875 ms
ConsoleHelper.ts:471 [Obsidian Linter] Running Compact YAML
ConsoleHelper.ts:471 compact-yaml: 0.27978515625 ms
ConsoleHelper.ts:471 [Obsidian Linter] Running Consecutive blank lines
ConsoleHelper.ts:471 consecutive-blank-lines: 0.410888671875 ms
ConsoleHelper.ts:471 convert-spaces-to-tabs: 0.079833984375 ms
ConsoleHelper.ts:471 [Obsidian Linter] Running Empty Line Around Blockquotes
ConsoleHelper.ts:471 empty-line-around-blockquotes: 0.320068359375 ms
ConsoleHelper.ts:471 [Obsidian Linter] Running Empty Line Around Code Fences
ConsoleHelper.ts:471 empty-line-around-code-fences: 0.606201171875 ms
ConsoleHelper.ts:471 empty-line-around-math-blocks: 0.064208984375 ms
ConsoleHelper.ts:471 [Obsidian Linter] Running Empty Line Around Tables
ConsoleHelper.ts:471 empty-line-around-tables: 2.490234375 ms
ConsoleHelper.ts:471 [Obsidian Linter] Running Heading blank lines
ConsoleHelper.ts:471 heading-blank-lines: 0.6181640625 ms
ConsoleHelper.ts:471 [Obsidian Linter] Running Line Break at Document End
ConsoleHelper.ts:471 line-break-at-document-end: 0.262939453125 ms
ConsoleHelper.ts:471 move-math-block-indicators-to-their-own-line: 0.060791015625 ms
ConsoleHelper.ts:471 paragraph-blank-lines: 0.085205078125 ms
ConsoleHelper.ts:471 [Obsidian Linter] Running Remove Empty Lines Between List Markers and Checklists
ConsoleHelper.ts:471 remove-empty-lines-between-list-markers-and-checklists: 1.248046875 ms
ConsoleHelper.ts:471 [Obsidian Linter] Running Remove link spacing
ConsoleHelper.ts:471 remove-link-spacing: 0.396240234375 ms
ConsoleHelper.ts:471 remove-space-around-characters: 0.078857421875 ms
ConsoleHelper.ts:471 [Obsidian Linter] Running Space after list markers
ConsoleHelper.ts:471 space-after-list-markers: 0.559814453125 ms
ConsoleHelper.ts:471 space-between-chinese-japanese-or-korean-and-english-or-numbers: 0.072998046875 ms
ConsoleHelper.ts:471 [Obsidian Linter] Running Trailing spaces
ConsoleHelper.ts:471 trailing-spaces: 0.468994140625 ms
ConsoleHelper.ts:471 [Obsidian Linter] Running Custom Regex
ConsoleHelper.ts:471 custom regex: 0.220947265625 ms
ConsoleHelper.ts:471 post rules: 0.30712890625 ms
ConsoleHelper.ts:471 rule running: 29751.885986328125 ms
ConsoleHelper.ts:471 [Obsidian Linter] Running Custom Lint Commands
```

So here are the slow performing rules:
- remove-multiple-spaces: 23.617 seconds
- two-spaces-between-lines-with-content: 6.1seconds

These two rules took up 99.88% of the time on this rule. It could be that the regex to ignore tables for the rule is causing it to take so long. I will check and see what other information I can find out about them.

**Comment by pjkaufman** [comment]
According to further investigation, the problem is ignoring tables. I don't really have a total fix for this. 
According to the timing info I added to determine what was going on for the rules in question:
```
ConsoleHelper.ts:471 [Obsidian Linter] Running Remove Multiple Spaces
ConsoleHelper.ts:471 table ignore: 23692.683837890625 ms
ConsoleHelper.ts:471 ignore type: 23695.718994140625 ms
ConsoleHelper.ts:471 text replace: 0.203125 ms
ConsoleHelper.ts:471 remove-multiple-spaces: 23697.48486328125 ms
ConsoleHelper.ts:471 [Obsidian Linter] Running Two Spaces Between Lines with Content
ConsoleHelper.ts:471 table ignore: 6254.947998046875 ms
ConsoleHelper.ts:471 two spaces: 0.1650390625 ms
ConsoleHelper.ts:471 two-spaces-between-lines-with-content: 6256.455810546875 ms
```

I am able to get the parsing time down to about 7 seconds for this exact instance by making table parsing happen last for `remove-multiple-spaces`.

**Comment by pjkaufman** [comment]
However the root cause of the issue does go unaddressed which is that for some reason the table regex is pretty slow to parse the table. I will have to look into this when I have more time. I will try to make this info easier for non-devs to submit  as a bug report in the future. Once I have that in place I will circle back around to see if anything can be done for things like this.

**Comment by pjkaufman** [comment]
These messages were accidentally added to the wrong issue as I had both open.

TLDR;
The issue is table parsing which I have little control over the parsing speed of. I got the timing down to 6 seconds for the file in question, but it will not help when the table is not in a codeblock.

**Comment by jerviscui** [comment]
I recently contracted COVID-19.

I remember having to upload `data.json` but it failed.
Then I may have been interrupted by other work.

**Comment by jerviscui** [comment]
[data.zip](https://github.com/platers/obsidian-linter/files/10265189/data.zip)

**Comment by jerviscui** [comment]
> TLDR； 问题是表解析，我无法控制它的解析速度。我将相关文件的时间缩短到 6 秒，但是当表格不在代码块中时它无济于事。

Yes, it's only the table that's in the code block that's a problem.
For me, the table is the output of a program, so I put it in a block of code.

**Comment by pjkaufman** [comment]
Given that at this time I am not really sure how to speed up the table recognition because it will either take 76 minutes for large files or 30 seconds for some smaller files, I am going to close this issue with the reorder of ignore types in this case. I am open to suggestions on how to speed this up though.
